libs/azure-api-client/src/ApiClient.php

```diff
@@ -107,6 +112,17 @@ public function __construct(
         ]);
     }
 
+    public function authenticate(string $resource): void
+    {
+        $middleware = Middleware::mapRequest(new AuthorizationHeaderResolver(
+            $this->authenticator,
+            $resource
+        ));
```

Comment:
Ten AuthorizationHeaderResolver je podle mě špatně nejde to vůbec rozšířit azure nepoužívá všude Bearer header, ale různé věci v tom mojém PR je AuthorizationHeaderResolverInterface navázaný na ten AuthorizationInterface aby to šlo rozšířit.